src/main.cpp

```diff
@@ -861,7 +861,7 @@ bool ContextualCheckTransaction(
     assert(!nu5Active || canopyActive);         // NU5 cannot be active unless Canopy is
     assert(!futureActive || nu5Active);         // ZFUTURE must include consensus rules for all supported network upgrades.
 
-    auto const orchard_bundle = tx.GetOrchardBundle();
+    auto& orchard_bundle = tx.GetOrchardBundle();
```

Comment:
I ❤️‍🔥 C++. We have been bitten by this particular issue multiple times now; is there maybe some kind of C++ linter we can run to find these?

I think we still want `const` here, because it's an error to try and modify the transaction's Orchard bundle here:
```suggestion
    const auto& orchard_bundle = tx.GetOrchardBundle();
```

Reply:
`auto&` will infer a reference to `const OrchardBundle` here.

Reply:
In that case, I'm fine leaving it as `auto&`. And we should get into the habit of using `auto&` everywhere we can (I presume the compiler will complain if we try to use it somewhere we can't, vs the reverse where it doesn't).

Reply:
Yes that's correct. In this case it also complains if you write just `OrchardBundle& orchard_bundle` because that casts away const.